python/ray/autoscaler/updater.py

```diff
@@ -58,13 +58,26 @@ def run(self,
             port_forward_cmd = self.kubectl + [
                 "port-forward",
                 self.node_id,
-            ] + [str(fwd) for fwd in port_forward]
+            ] + [
+                "{}:{}".format(local, remote) for local, remote in port_forward
+            ]
             port_forward_process = subprocess.Popen(port_forward_cmd)
-            # Give port-forward a grace period to run and print output before
-            # running the actual command. This is a little ugly, but it should
-            # work in most scenarios and nothing should go very wrong if the
-            # command starts running before the port forward starts.
-            time.sleep(1)
+            try:
+                # Give port-forward a chance to connect before running the
+                # actual command. If port forward fails (and returns
+                # immediately) we failed to port forward and should abort the
+                # command
+                port_forward_process.wait(5)
```

Comment:
This seems like a very long time to wait before launching the other process. This should fail relatively fast, no?